fix issue #1380 - remove ascii greeting from all serverless CLI comma…

bennybauer opened the pull request:
##### Status:
Ready

##### Description:
* Removed ascii greeting from all displayHelp calls
* Added the greeting to create.finish()
* Added test to create.js
* **Didn't** remove the reference to the docs mentioned in #1380 as I find it useful for now. WDYT?

flomotlik commented:
@bennybauer could you rebase from the v1.0 branch? We've now disabled integration tests for forks so this should fix it then. Looking into the PR

flomotlik commented on lib/classes/CLI.js:
```diff
@@ -60,23 +60,23 @@ class CLI {
     }
+
+    // if only "help" or "h" was entered
+    if ((commands.length === 0) ||
```
Any particular reason why this was moved down here?

flomotlik replied:
Probably so the two help sections are next to each other. Makes sense

bennybauer replied:
If this line will be checked first we'll never get to check for the version option (or other options). I thought it will be a cleaner way to do that without adding more checks to this condition.

flomotlik commented:
@pmuens this looks good to me. Any additional thoughts? Imho ready to be merged when its rebased and the build passes.

pmuens commented on lib/classes/CLI.js:
```diff
@@ -9,7 +9,7 @@ const chalk = require('chalk');
   constructor(serverless, inputArray) {
     this.serverless = serverless;
-    this.inputArray = (typeof inputArray !== 'undefined' ? inputArray : []);
+    this.inputArray = inputArray || undefined;
```
Would be better to use `null` here because `undefined` is a more JavaScript engine specific feature which a variable shouldn't be set to.

pmuens commented on tests/classes/CLI.js:
```diff
@@ -27,9 +27,9 @@ describe('CLI', () => {
     });
 
-    it('should set an empty inputArray when none is provided', () => {
+    it('should set an undefined inputArray when none is provided', () => {
```
This test needs a tiny update once the variable setting from `undefined` to `null` is done.

pmuens commented:
Yep. Added two comments. Other than that I think it's GTM. Thanks @bennybauer

bennybauer commented:
Thanks for the comments guys, I'll get to it in a day or two.

bennybauer commented:
I had some git hell, so I've created 2 separate PRs for the issues mentioned above, addressing the review: PR #1579 and PR #1580. Feel free to close this PR.

Thanks!

flomotlik commented:
Ok cool, closing this one and looking into those two PR's
